src/backend/utils/adt/ruleutils.c

```
@@ -549,6 +550,73 @@ static void get_json_table_nested_columns(TableFunc *tf, JsonTablePlan *plan,
 
 #define only_marker(rte)  ((rte)->inh ? "" : "ONLY ")
 
+Datum
+pg_get_roledef(PG_FUNCTION_ARGS)
+{
+	Oid			roleid;
+	HeapTuple	tuple;
+	Form_pg_authid roleform;
+	StringInfoData buf;
+	char	   *rolname;
+	Datum		rolvaliduntil_datum;
+	bool		rolvaliduntil_isnull;
+
+	roleid = PG_GETARG_OID(0);
+	if (!OidIsValid(roleid))
+		PG_RETURN_NULL();
+	/* Look up the role in pg_authid */
+	tuple = SearchSysCache1(AUTHOID, ObjectIdGetDatum(roleid));
+	if (!HeapTupleIsValid(tuple)) /* Role not found */
+		PG_RETURN_NULL();
```

would you need an `ereport` for this one? If the role isn't found, then the user didn't create it. That would be a user issue, right? I am thinking that it might be useful for the user to be notified that they didn't create a role, and not that the function just failed for an undetermined reason.

---

What you say is true, *but* this function takes an argument of type regrole, which is going to resolve the role name to OID, and if that role name doesn't exist, then it's going to raise an error, so you won't ever get to this point for a role that doesn't exist, except if it's passed as OID instead of by name.

---

(actually this makes me think that perhaps it would be more consistent to throw errors rather than return null for the cases where the user doesn't exist)